### DISCUSSION TITLE
I would really really like to use Pixi as a complete Conda replacement for production pipelines, but...

### DISCUSSION THREAD

**Discussion by jwhite007** [prefix-dev/pixi#1444] [General]
I'm a Bioinformatics scientist and manage the executables framework at my workplace. I currently use Conda to provide isolated environments for various production pipelines which have different dependencies. Apparently Pixi's theme is development with the project being the central focus. One has to enter into a project directory before one can enter into a particular environment shell. With Conda, we can activate an environment and process data no matter the cwd. I haven't yet seem how I can configure Pixi to work similarly. Is it even possible? 

In addition, I've found "Pixi search" a bit limiting. There's no way to show, for instance, all Python versions available on a particular channel. It's only the latest that is shown.

What I like about Pixi is that it's built with Rust, that it's very speedy, and that it allows for installation of applications globally; however, its theme is development. It would be nice to have a similar tool focused on production.

**Comment by synapticarbors** [comment]
The issue you've run into with pixi being project based vs conda's management of a global env registry is one that has been discussed a number of times, but there is a particularly lengthy one here:
https://github.com/prefix-dev/pixi/discussions/1021

Currently the best, although not ideal, option is to use the `--manifest-path` flag. So for example you could activate a pixi env by running from any cwd: `pixi shell --manifest-path <path to project toml file>`. The good thing is that in recent versions of pixi, that environment is persistent across future calls to `pixi` as long as you remain in that shell env. 

The downside of specifying the manifest path is that it's verbose and requires you to know where that file is located on your system. On the prefix.dev discord, someone posted a shell script a while back with an example of how to create an activate like command:
https://discord.com/channels/1082332781146800168/1224840062588686479/1225035708989702175

**Comment by jwhite007** [comment]
@synapticarbors,

Thank you for the detailed response. I appreciate the links and suggestions. I'll be following https://github.com/prefix-dev/pixi/discussions/1021. Hopefully, in the future as Pixi gets more fleshed out, it will be able to handle my use case in a more streamlined manner.

**Comment by Ethan-Hetrick** [comment]
Call it jank -- a colleague showed me this way to activate a pixi project as a conda environment, and it works well for bridging the old scripts we have laying around that still use `conda activate`. Long-term, migrate to project-level environment management or containerize your pixi environments with Singularity/Docker (unless there is a better best-practice I'm not aware of)

```bash
# Use conda to activate the pixi environment
conda activate ~/test-pixi/.pixi/envs/default # Hidden dir inside pixi project
```

**Comment by pavelzw** [comment]
> In addition, I've found "Pixi search" a bit limiting. There's no way to show, for instance, all Python versions available on a particular channel. It's only the latest that is shown

this should have been improved in pixi 0.66.0:

```
pixi search python --limit -1
pixi search --json
```

there is also [pixi-browse](https://github.com/pavelzw/pixi-browse) which allows you to view details of all conda packages in a specific channel:

```
pixi exec pixi-browse --matchspec python -c conda-forge
```
